File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return None
        elif root.left == None and root.right == None:
            return root
        elif root.left == None and root.right != None:
            right = self.invertTree(root.right)
            root.right = None
            root.left = right
            return root
        elif root.left != None and root.right == None:
            left = self.invertTree(root.left)
            root.left = None
            root.right = left
            return root
        elif root.left != None and root.right != None:
            left = self.invertTree(root.left)
            right = self.invertTree(root.right)
            root.left = right
            root.right = left
            return root

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        if (root == null){
            return new ArrayList<Integer>();
        }
        List<Integer> result = new ArrayList<>();
        if (root.left != null){
            List<Integer> leftTree = postorderTraversal(root.left);
            result.addAll(leftTree);
        }
        if (root.right != null){
            List<Integer> rightTree = postorderTraversal(root.right);
            result.addAll(rightTree);
        }
        result.add(root.val);
        return result;
    }
}